quickwit-metastore/src/lib.rs

```diff
@@ -59,3 +63,86 @@ pub use metastore::{
     IndexMetadata, MetadataSet, Metastore, SplitMetadata, SplitMetadataAndFooterOffsets, SplitState,
 };
 pub use metastore_resolver::{MetastoreFactory, MetastoreUriResolver};
+
+/// A trait to allow for our data model to check itself
+#[async_trait]
+pub trait Check {
+    /// Something that can be implemented
```

Comment:
This is not a very good comment, and possibly not a good name.
When I first read "description", I was expecting either a description of the resource that was checked, or a description of what was checked.

Looking at the URI of the metastore it is neither: 
You just return the URI for the metastore for instance.

```
/// A concise description resuming the setting that were used to build an instance of 
/// this resource. The type of the resource (e.g. Metastore) does not need to be repeated. 
/// If the resource is built from an URI, returning just the URI is best.
```

Reply:
You are right. I had thought more information would be beneficial here, but later reverted that and just added `&'static str`s to the `check_all!` macro. I'm going to remove this function.

Reply:
URI is good to identify what has been checked, I think it's also important to add the object type too. Something like this:
- For a storage:  `Storage - s3://my-bucket/indexes: ✔️`
- For a metastore:  `Metastore - file:///my-metastore-dir: ✔️`
- For a source:  `Source - file:///my-metastore-dir: ✔️`

Reply:
@fmassot the object type is added by the calling function.